src/main/java/seedu/cafectrl/storage/Decoder.java

```diff
@@ -7,9 +7,50 @@
 import seedu.cafectrl.ui.Ui;
 
 import java.util.ArrayList;
+import java.util.Arrays;
 
+
+//from txt file to arraylist
 public class Decoder {
     private static final Ui ui = new Ui();
+
+    /**
+     * Decodes an ArrayList of string lines into a Menu object, reconstructing its content.
+     *
+     * @param textLines An ArrayList of strings representing the encoded Menu data.
+     * @return A Menu object containing the decoded Menu data.
+     */
+    public static Menu decodeMenuData(ArrayList<String> textLines) {
+        ArrayList<Dish> menuDishList = new ArrayList<>();
+        for(String dishString : textLines){
```

Comment:
Perhaps add a whitespace after for loop conditional
```suggestion
        for(String dishString : textLines) {
```